addons/base_import/models/base_import.py

```diff
@@ -476,107 +554,208 @@ def _try_match_date_time(self, preview_values, options):
+            for field in fields_tree:
+                # FIXME: should match all translations & original
+                string_match = None
+                if header.lower() == field['name'].lower():
+                    return {
+                        'field_path': [field['name']],
+                        'distance': 0
+                    }
+                field_string = field.get('string', '').lower()
+                if header.lower() == field_string:
+                    # matching string are not reliable way because
+                    # strings have no unique constraint
+                    string_match = field
+                translated_header = IrTranslation._get_source('ir.model.fields,field_description', 'model', self.env.lang, header).lower()
+                if translated_header == field_string:
+                    string_match = field
+                if string_match:
+                    # this behavior is only applied if there is no matching field['name']
+                    return {
+                        'field_path': [string_match['name']],
+                        'distance': 0
+                    }
+
+            # If no match found, try fuzzy match on most likely fields (based on header type)
+            # Filter out fields with types that does not match corresponding header types.
+            most_likely_fields = self._get_most_likely_fields(fields_tree, header_types)
+            if not most_likely_fields:
+                return {}
+
+            fields_matching_distance = {}
+            for field in most_likely_fields:
+                # use string distance for fuzzy match only on most likely field types
+                name_field_dist = Levenshtein.distance(header.lower(), field['name'].lower()) / max(len(field['name'].lower()), len(header.lower()))
+                string_field_dist = Levenshtein.distance(header.lower(), field_string) / max(len(field_string), len(header.lower()))
+                fields_matching_distance[field['name']] = min([name_field_dist, string_field_dist])
+
+            # Keep only the closest mapping suggestion. Note that in case of multiple mapping on the same field,
+            # a mapping suggestion could be canceled by another one that has a smaller distance on the same field.
+            # See 'deduplicate_mapping_suggestions' method for more info.
+            min_dist_field = min(fields_matching_distance, key=fields_matching_distance.get)
+            distance = fields_matching_distance[min_dist_field]
+            if distance < self.FUZZY_MATCH_DISTANCE:
+                return {
+                    'field_path': [next(field['name'] for field in most_likely_fields if field['name'] == min_dist_field)],
+                    'distance': distance
+                }
+            else:
+                return {}
 
         # relational field path
-        traversal = []
-        subfields = fields
+        hierarchy_map = []
```

Comment:
As Thibault mentioned, it's strange that a variable named "_map" is actually a list.
I mean it's ALSO simply a path to the field even though it has multiple layers, so why not simply "field_path" ?

Reply:
allez, vendu (and it would be aligned with the field_path I use further in the process)